### PR TITLE
Rework PyQt cmake finders

### PR DESCRIPTION
**Pull request by martyngigg**
**Description of work.**

This is some work in preparation for the new workbench. It adds a `FindPyQt5` cmake finder. The code in the original `FindPyQt4` has been refactored to be able to share it between both modules.

**To test:**

Code review.
The current code should build.

*No issue number*

**Release Notes** 
*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: buildconfig/CMake/FindPyQt5.cmake [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Find PyQt5
+# ~~~~~~~~~~
+#
+# PyQt4 website: http://www.riverbankcomputing.co.uk/pyqt/index.php
+#
+# Find the installed version of PyQt4. FindPyQt5 should only be called after
```
Comment (soininen) [inline buildconfig/CMake/FindPyQt5.cmake]: PyQt4 -> PyQt5

File: buildconfig/CMake/PyQtFindImpl.cmake [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+#
+# PYQT${major_version}_VERSION_TAG - The Qt version tag used by PyQt's sip files.
+#
+# PYQT${major_version}_SIP_DIR - The directory holding the PyQt .sip files.
+#
+# PYQT${major_version}_SIP_FLAGS - The SIP flags used to build PyQt.
+function (find_pyqt major_version)
+  if (PYQT${major_version}_VERSION)
+    # Already in cache, be silent
+    set( PYQT${major_version}_FOUND TRUE)
```
Comment (soininen) [inline buildconfig/CMake/PyQtFindImpl.cmake]: Extra space after `set(`.

File: buildconfig/CMake/PyQtFindImpl.cmake [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  else ()
+      find_file (_find_pyqt_py FindPyQt.py PATHS ${CMAKE_MODULE_PATH})
+      execute_process (COMMAND ${PYTHON_EXECUTABLE} ${_find_pyqt_py} ${major_version} OUTPUT_VARIABLE _pyqt_config)
+      if(_pyqt_config)
+        string (REGEX MATCH "^pyqt_version:([^\n]+).*$" _dummy ${_pyqt_config})
+        set (PYQT${major_version}_VERSION "${CMAKE_MATCH_1}" CACHE STRING "PyQt4's version as a 6-digit hexadecimal number")
```
Comment (soininen) [inline buildconfig/CMake/PyQtFindImpl.cmake]: PyQt4 -> PyQt5 here and similarly below.

**Review by soininen — CHANGES_REQUESTED** [review]
Looks good except for the few typos and missing PyQt4 -> PyQt5 replaces.

**Comment by martyngigg** [comment]
Thanks for spotting those @soininen. I've tidied those up now.

**Review by soininen — APPROVED** [review]

**Pull request merged**